From 1755a9bfb3e87f2503d3f1203d00e1f3b1be26c0 Mon Sep 17 00:00:00 2001
From: Sunita Nadampalli <sunitan@ti.com>
Date: Fri, 11 Mar 2011 10:00:46 -0600
Subject: OMAP Overlay: Fix for overlay window checks against panel timings

Description# Currently we are checking for overlay input window parameters
against panel resolution for parameter validity check. But ideally the check
should be against overlay output window. This logic is corrected inside manager
store method.

Validation# AV playback on HDMI and LCD
pathset4# Modified printk statement to be more meaningful.

Change-Id: I17306aae7d95ca848c99c219efd337cd41932d1f
Signed-off-by: Sunita Nadampalli <sunitan@ti.com>
---
 drivers/video/omap2/dss/overlay.c |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

Index: kernel/drivers/video/omap2/dss/overlay.c
===================================================================
--- kernel.orig/drivers/video/omap2/dss/overlay.c	2011-04-05 21:57:06.000000000 +0900
+++ kernel/drivers/video/omap2/dss/overlay.c	2011-04-06 00:30:44.000000000 +0900
@@ -90,6 +90,7 @@
 
 	if (mgr && sysfs_streq(mgr->name, "tv")) {
 		ovl->get_overlay_info(ovl, &info);
+		
 		if (mgr->device->panel.timings.x_res < info.out_width ||
 			mgr->device->panel.timings.y_res < info.out_height) {
 			printk(KERN_ERR"output window size exceeds panel dimensions");
